src/OmnichannelChatSDK.ts

```diff
@@ -1037,9 +1109,17 @@ class OmnichannelChatSDK {
                 this.authSettings = authSettings;
             }
 
-            const {PreChatSurvey: preChatSurvey, msdyn_prechatenabled, msdyn_callingoptions, msdyn_conversationmode} = liveWSAndLiveChatEngJoin;
+            const {PreChatSurvey: preChatSurvey, msdyn_prechatenabled, msdyn_callingoptions, msdyn_conversationmode, msdyn_enablechatreconnect} = liveWSAndLiveChatEngJoin;
             const isPreChatEnabled = msdyn_prechatenabled === true || msdyn_prechatenabled == "true";
 
+            if (this.authSettings) {
```

Comment:
Please move this if block after the `getAuthToken()` has been called successfully. It could happen that `getAuthToken()` fails, then it should be treated as an unauthenticated chat.